### PR TITLE
Add do_transform_polygon_stamped

### PR DESCRIPTION
**Pull request by tonynajjar**
https://github.com/ros2/geometry2/issues/581

This is an initial draft, please give feedback on the idea, thanks!

### CODE REVIEW

File: tf2_geometry_msgs/src/tf2_geometry_msgs/tf2_geometry_msgs.py [hunk 1]
```diff
@@ -31,9 +31,9 @@
 
 from typing import Iterable, Optional, Tuple
 
-from geometry_msgs.msg import (PointStamped, Pose, PoseStamped,
-                               PoseWithCovarianceStamped, TransformStamped,
-                               Vector3Stamped)
+from geometry_msgs.msg import (PointStamped, Point, Point32, PolygonStamped, Pose,
+                               PoseStamped, PoseWithCovarianceStamped,
+                               TransformStamped, Vector3Stamped)
```
Comment (clalancette) [inline tf2_geometry_msgs/src/tf2_geometry_msgs/tf2_geometry_msgs.py]: flake8 suggests:

```suggestion
from geometry_msgs.msg import (Point, Point32, PointStamped, PolygonStamped, Pose,
                               PoseStamped, PoseWithCovarianceStamped,
                               TransformStamped, Vector3Stamped)
```

File: tf2_geometry_msgs/src/tf2_geometry_msgs/tf2_geometry_msgs.py [hunk 2]
```diff
@@ -409,3 +409,31 @@ def do_transform_pose_with_covariance_stamped(
 
 tf2_ros.TransformRegistration().add(PoseWithCovarianceStamped,
                                     do_transform_pose_with_covariance_stamped)
+
```
Comment (clalancette) [inline tf2_geometry_msgs/src/tf2_geometry_msgs/tf2_geometry_msgs.py:412]: Add an extra newline here:

```suggestion


```

**Review by clalancette — CHANGES_REQUESTED** [review]
In short, this is completely new functionality so I'm happy to have it in.  There are two flake8 errors to be resolved, but once those are fixed we can run CI on it.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18254/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12790/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18929/)

**Comment by tonynajjar** [comment]
Would love to have it backported to Humble if possible

**Comment by clalancette** [comment]
@Mergifyio backport humble

**Pull request merged**